rotkehlchen/chain/evm/decoding/metamask/decoder.py

```diff
@@ -81,7 +81,10 @@ def _decode_swap(self, context: DecoderContext) -> DecodingOutput:
                 event.notes = f'Swap {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} in metamask'  # noqa: E501
                 event.address = self.router_address
                 out_event = event
-            elif event.event_type == HistoryEventType.RECEIVE and event.location_label == sender:  # find the receive event  # noqa: E501
+            elif (
+                    (event.event_type == HistoryEventType.TRADE and event.event_subtype == HistoryEventSubType.RECEIVE) or  # noqa: E501
+                    (event.event_type == HistoryEventType.RECEIVE and event.event_subtype == HistoryEventSubType.NONE and event.location_label == sender)  # noqa: E501
+            ):  # find the receive event
```

Comment:
Added this to fix a metamask test. Metamask apparently routes through openocean sometimes, in which case the event has already been modified to trade/receive here by the openocean decoder. It's already checking for trade/spend, so no need to change that.

Reply:
Can you expain this in a comment too?